src/main/java/com/marklogic/kafka/connect/source/RowBatcherSourceTask.java

```diff
@@ -82,11 +74,23 @@ public List<SourceRecord> poll() throws InterruptedException {
         return newSourceRecords.isEmpty() ? null : newSourceRecords;
     }
 
-    protected RowBatcher<JsonNode> newRowBatcher(List<SourceRecord> newSourceRecords) {
-        ContentHandle<JsonNode> jsonHandle = new JacksonHandle().withFormat(Format.JSON).withMimetype("application/json");
-        rowBatcher =  dataMovementManager.newRowBatcher(jsonHandle);
-        configureRowBatcher(parsedConfig, rowBatcher, newSourceRecords);
-        return rowBatcher;
+    protected RowBatcher<?> newRowBatcher(List<SourceRecord> newSourceRecords) throws Exception {
+        MarkLogicSourceConfig.OUTPUT_TYPE outputType = MarkLogicSourceConfig.OUTPUT_TYPE.valueOf((String) parsedConfig.get(MarkLogicSourceConfig.OUTPUT_FORMAT));
+        final Supplier<RowBatcher<?>> rowBatcherSupplier;
+        switch (outputType) {
+            case JSON :
+                rowBatcherSupplier = new JsonRowBatcherSupplier(dataMovementManager, newSourceRecords, parsedConfig);
+                break;
+            case XML:
+                rowBatcherSupplier = new XmlRowBatcherSupplier(dataMovementManager, newSourceRecords, parsedConfig);
+                break;
+            case CSV:
+                rowBatcherSupplier = new CsvRowBatcherSupplier(dataMovementManager, newSourceRecords, parsedConfig);
+                break;
+            default:
+                throw new Exception("");
```

Comment:
We don't expect this, but in the exceedingly rare event that it does happen, a nice message is still helpful - I would do `throw new IllegalArgumentException("Unexpected output type: " + outputType);` . That also extend RuntimeException so you don't have to declare `throws Exception` on this method.

Reply:
Thanks, meant to come back to this, but forgot.